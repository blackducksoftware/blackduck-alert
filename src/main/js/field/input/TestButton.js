import React from 'react';
import PropTypes from 'prop-types';

const TestButton = ({ onClick, children }) => {
    return (
        <button className="btn btn-primary" type="button" onClick={onClick}>{children}</button>
    );
};

TestButton.defaultProps = {
    children: 'Test'
};

TestButton.propTypes = {
    children: PropTypes.string,
    onClick: PropTypes.func.isRequired
};

export default TestButton;