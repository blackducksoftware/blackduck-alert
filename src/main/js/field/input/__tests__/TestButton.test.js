import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import TestButton from '../TestButton';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

test('Rendering default test button snapshot', () => {
    const button = renderer.create(
        <TestButton onClick={()=> {}} />
    );
    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
});

test('TestButton click handler', () => {
    const mockCallBack = jest.fn();
    const button = shallow((<TestButton onClick={mockCallBack} />));

    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);

    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(2);
});

test('TestButton alt label', () => {
    const mockCallBack = jest.fn();
    const button = shallow((<TestButton onClick={mockCallBack}>Test Config</TestButton>));

    expect(button.text()).toEqual('Test Config');
    console.log(button.text());
});
