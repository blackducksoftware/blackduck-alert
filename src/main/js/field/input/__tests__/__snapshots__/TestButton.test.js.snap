// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering default test button snapshot 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;
